firebase/authentication.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  browserLocalPersistence,
+  GoogleAuthProvider,
+  setPersistence,
+  signOut,
+  signInWithPopup,
+} from "firebase/auth";
+import { auth } from "./firebase";
+import type { User } from "firebase/auth";
+
+export function getCurrentUser(): User | null {
+  return auth.currentUser;
+}
+
+auth.onAuthStateChanged(() => {
+  const user = getCurrentUser();
+  if (user) {
+    console.log("User is logged in");
+  } else {
+    console.log("User is logged out");
+  }
+});
+
+export async function signInUserWithGoogle(): Promise<void> {
+  const provider = new GoogleAuthProvider();
+  await setPersistence(auth, browserLocalPersistence);
+  try {
+    const result = await signInWithPopup(auth, provider);
+    const credential = GoogleAuthProvider.credentialFromResult(result);
+    console.log("credential: ", credential);
```

Comment:
...And we should definitely not share credentials with Vercel

Reply:
Has been removed